Tasks/google-play-release/GooglePlay.ts

```diff
@@ -42,54 +42,86 @@ interface GlobalParams {
             } else {
-                console.error(tl.loc('InvalidAuthFile'));
-                throw new Error(tl.loc('InvalidAuthFilewithName', serviceAccountKeyFile));
+                tl.debug(`The service account file path ${serviceAccountKeyFile} points to a directory.`);
+                throw new Error(tl.loc('InvalidAuthFile', serviceAccountKeyFile));
             }
         } else if (authType === 'ServiceEndpoint') {
-            let serviceEndpoint: tl.EndpointAuthorization = tl.getEndpointAuthorization(tl.getInput('serviceEndpoint', true), true);
-            if (!serviceEndpoint) {
-                throw new Error(tl.loc('EndpointNotFound'));
-            }
+            let serviceEndpoint: tl.EndpointAuthorization = tl.getEndpointAuthorization(tl.getInput('serviceEndpoint', true), false);
             key.client_email = serviceEndpoint.parameters['username'];
             key.private_key = serviceEndpoint.parameters['password'].replace(/\\n/g, '\n');
         }
 
-        let apkFile: string = resolveGlobPath(tl.getPathInput('apkFile', true));
-        let apkFileList: string[] = [apkFile];
-        let additionalApks: string[] = tl.getDelimitedInput('additionalApks', '\n');
-        if (additionalApks.length > 0) {
-            for (let i = 0; i < additionalApks.length; i++) {
-                apkFileList.push(resolveGlobPath(additionalApks[i]));
-            }
+        let mainApkPattern = tl.getPathInput('apkFile', true);
+        tl.debug(`Main APK pattern: ${mainApkPattern}.`);
+
+        let apkFile: string = resolveGlobPath(mainApkPattern);
+        tl.checkPath(apkFile, 'apkFile');
+        let mainVersionCode = apkParser.readFile(apkFile).readManifestSync().versionCode;
+        tl.debug(`    Found the main APK file: ${apkFile}, versionCode: ${mainVersionCode}.`);
+
+        let apkFileList: string[] = getAllApkPaths(apkFile);
+        if (apkFileList.length > 1) {
             console.log(tl.loc('FoundMultiApks'));
             console.log(apkFileList);
         }
 
+        let versionCodeFilterType: string = tl.getInput('versionCodeFilterType', false) ;
+        let versionCodeFilter: string | number[] = null;
+        if (versionCodeFilterType === 'list') {
+            versionCodeFilter = getVersionCodeListInput();
+        } else if (versionCodeFilterType === 'expression') {
+            versionCodeFilter = tl.getInput('replaceExpression', false);
+        }
+
         let track: string = tl.getInput('track', true);
         let userFraction: number = Number(tl.getInput('userFraction', false)); // Used for staged rollouts
         let changelogFile: string = tl.getInput('changelogFile', false);
-        let shouldAttachMetadata: any = JSON.parse(tl.getInput('shouldAttachMetadata', false));
+        let shouldAttachMetadata: boolean = tl.getBoolInput('shouldAttachMetadata', false);
+        let shouldUploadApks: boolean = !shouldAttachMetadata || tl.getBoolInput('shouldUploadApks', false);
```

Comment:
IMO that does not make sense here because the input id of type Boolean. If the user didn't check the check-box, that means false, but not undefined.